File: Algorithms/Medium/P1456-Maximum_Number_of_Vowels_in_a_Substring_of_Given_Length/p1456.ac.go
func maxVowels(s string, k int) int {
    isVowel := func(c byte) bool {
        if c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u' {
            return true
        }
        return false
    }

    front, end, vowel_counter := 0, k, 0

    for i := 0; i < k; i++ {
        if isVowel(s[i]) { vowel_counter++ }
    }

    max := vowel_counter

    for i := 0; i < len(s); i++ {
        if end >= len(s) { break }
        if isVowel(s[front]) { vowel_counter-- }
        if isVowel(s[end]) { vowel_counter++ }
        front++
        end++
        if max < vowel_counter { max = vowel_counter }
    }

    return max
}

File: Algorithms/Medium/P1456-Maximum_Number_of_Vowels_in_a_Substring_of_Given_Length/p1456.ac.cc
class Solution {
public:
    int maxVowels(string s, int k) {
        auto isVowel = [&](const char c) -> bool {
            if (c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u')
                return true;
            return false;
        };

        int front = 0, end = k, vowel_counter = 0;
        for (int i = 0; i < k; ++i) {
            if (isVowel(s[i])) ++vowel_counter;
        }

        int max = vowel_counter;

        for (int i = 0; i < s.length(); ++i) {
            if (end >= s.length()) break;
            if (isVowel(s[front++])) --vowel_counter;
            if (isVowel(s[end++])) ++vowel_counter;
            max = vowel_counter > max ? vowel_counter : max;
        }

        return max;
    }
};